### ISSUE TITLE
trapping a vehicle in a bubble airlock

### ISSUE THREAD

**Issue by sah-huawei** [huawei-noah/SMARTS#1064]
## **BUG REPORT**
SMARTS is trying to trap a vehicle that has just entered a bubble airlock on the same step.  This causes problems because both things cause sensors to be attached to the vehicle, which cannot be done twice.  

**SMARTS version**
0.5.0

**Steps to reproduce the bug**
Run the pytest in `test_rllib_hiway_env.py`.

 The `test_rllib_hiway_env` will fail with an assert like:
```
(pid=32731) ERROR:SMARTS:trip_meter_sensor already added
(pid=32731) Traceback (most recent call last):
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/smarts.py", line 165, in step
(pid=32731)     return self._step(agent_actions, time_delta_since_last_step)
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/smarts.py", line 230, in _step
(pid=32731)     self._trap_manager.step(self)
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/trap_manager.py", line 185, in step
(pid=32731)     sim, vehicle_id, agent_id, mission
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/trap_manager.py", line 247, in _hijack_vehicle
(pid=32731)     sim, vehicle_id, agent_id, agent_interface, plan
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/utils/cache.py", line 130, in wrapper
(pid=32731)     return func(self, *args, **kwargs)
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/vehicle_index.py", line 332, in start_agent_observation
(pid=32731)     Vehicle.attach_sensors_to_vehicle(sim, vehicle, agent_interface, plan)
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/vehicle.py", line 441, in attach_sensors_to_vehicle
(pid=32731)     plan=plan,
(pid=32731)   File "/home/steve/Repos/Huawei/SMARTS/smarts/core/vehicle.py", line 638, in attach_sensor
(pid=32731)     ), f"{sensor_name} already added to {self.id} {id(self)}"
(pid=32731) AssertionError: trip_meter_sensor already added
```

Note that this test uses the `loop` scenario, which defines a bubble to follow the `laner_actor` around and capture the `open_agent` actor.   Just before this assert, vehicle `car-flow-random-route-06af8414-3188849943547999844--2-0.0` has entered an airlock for this bubble and had sensors attached.  Now it is being trapped and having sensors attached again, which triggers the assert.

**Resulting and expected behaviour**
We should probably check the `vehicle_index` when considering vehicles to trap to ensure that they are not already in a bubble airlock ("shadowed").

**Comment by sah-huawei** [comment]
I added a fix for this to PR #1048.

**Comment by Gamenot** [comment]
I think that this behaviour is what we were seeing in: https://github.com/huawei-noah/SMARTS/issues/411
